features/net/network-socket/nsapi_dns.cpp

```diff
@@ -189,16 +190,22 @@ static int dns_scan_response(const uint8_t **p, nsapi_addr_t *addr, unsigned add
 }
 
 // core query function
-static int nsapi_dns_query_multiple(NetworkStack *stack,
-        nsapi_addr_t *addr, unsigned addr_count,
-        const char *host, nsapi_version_t version)
+static int nsapi_dns_query_multiple(NetworkStack *stack, const char *host,
+        nsapi_addr_t *addr, unsigned addr_count, nsapi_version_t version)
 {
     // check for valid host name
     int host_len = host ? strlen(host) : 0;
     if (host_len > 128 || host_len == 0) {
         return NSAPI_ERROR_PARAMETER;
     }
 
+    // check for simple ip addresses
+    SocketAddress address;
+    if (address.set_ip_address(host) && address.get_ip_version() == version) {
+        *addr = address.get_addr();
```

Comment:
If it's a valid literal but the version doesn't match, it should return an error rather than proceed to try to resolve it.